File: RiskAssessment/API/RiskManagement.Web/src/app/pages/correctiveaction/correctiveaction.model.ts
import { NgbDateParserFormatter, NgbDateStruct, NgbTimeStruct } from '@ng-bootstrap/ng-bootstrap';
import { LocationModel, TitleModel, SubTitleModel, Title, SubTitle, Action, EventActionStatusModel, events, EventFilterModel, EventViewModel, EventDataModel, EventFilterViewModel, EventAttendeeDataModel, ActionDataModel, ActionResponsiblePersonDataModel, AgendaDataModel } from '../calendar/formeventdata';
export class CorrectiveActionDataModel {
    public ActionID: number;
    public CorrectiveActionId: number;
    public Title: string;
    public AssignedBy: number;
    public CategoryID: number;
    public ActionSourceID: number;
    public ActionTaken: string;
    public Duedate: string;
    public dueDateStruct: NgbDateStruct;
    public RootCause: string;
    public CorrectiveActionNeeded: boolean;
    public IdentifiedCorrectiveAction: string;
    public RiskLevel: number;
    public ResponsibleParty: number;
    public CorrectiveActionDueDate: string;
    public CorrectiveActionDueDateStruct: NgbDateStruct;
    public IssueResolved: boolean;
    public DateResolved: string;
    public dateResolved: NgbDateStruct;
    public resultList: string[];
    public ResultList: string[];
    public ActionResponsiblePersonDataModel: ActionResponsiblePersonDataModel[];
    public actionResponsiblePersonDataModel: ActionResponsiblePersonDataModel[];
    constructor() {
    }
}

export class CorretiveActionListPageFilterModel {
    public PageNo: number;
    public PageSize: number;
    public SortColumn: string;
    public SortOrder: string;
    public TitleFilter: string;
    public AssignedByFilter: string;
    public CategoryFilter: string;
    public SourceFilter: string;
    public ActionTakenFilter: string;
    public DueDateFilter: string;
    public RootCauseFilter: string;
}

export class CorrectiveActionListModel {
    public CorrectiveActionId: number;
    public Title: string;
    public AssignedBy: string;
    public CategoryName: string;
    public ActionSource: string;
    public ActionTaken: string;
    public DueDate: string;
    public RootCause: string;
    public RowID: number;
}

export class CorrectiveActionList {

    public CorrectiveActionListModel: CorrectiveActionListModel[];
    public TotalRecords: number;
}

File: Angular2_AspDotNet/API/Angular2_AspDotNet.Models/CorrectiveActionModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Angular2_AspDotNet.Models
{
    public class CorrectiveActionModel
    {
        public int CorrectiveActionId { get; set; }
        public string CorrectiveActionName { get; set; }
    }

    public class CorrectiveActionDataModel
    {
        public CorrectiveActionDataModel()
        {
            this.resultList = new List<string>();
            this.ActionResponsiblePersonDataModel = new List<Models.ActionResponsiblePersonDataModel>();
        }
        public int ActionID { get; set; }
        public int CorrectiveActionId { get; set; }
        public string Title { get; set; }
        public Nullable<int> AssignedBy { get; set; }
        public Nullable<int> CategoryID { get; set; }
        public Nullable<int> ActionSourceID { get; set; }
        public string ActionTaken { get; set; }
        public DateTime Duedate { get; set; }
        public string RootCause { get; set; }

        public Nullable<bool> CorrectiveActionNeeded { get; set; }
        public string IdentifiedCorrectiveAction { get; set; }
        public Nullable<int> RiskLevel { get; set; }
        public Nullable<int> ResponsibleParty { get; set; }
        public Nullable<System.DateTime> CorrectiveActionDueDate { get; set; }
        public Nullable<bool> IssueResolved { get; set; }
        public Nullable<System.DateTime> DateResolved { get; set; }
        public List<string> resultList { get; set; }
        public List<string> ResultList { get; set; }
        public List<ActionResponsiblePersonDataModel> ActionResponsiblePersonDataModel { get; set; }

    }
    public class CorretiveActionListPageFilterModel
    {
        public int PageNo { get; set; }
        public int PageSize { get; set; }
        public string SortColumn { get; set; }
        public string SortOrder { get; set; }

        public string TitleFilter { get; set; }
        public string AssignedByFilter { get; set; }
        public string CategoryFilter { get; set; }
        public string SourceFilter { get; set; }
        public string ActionTakenFilter { get; set; }
        public string DueDateFilter { get; set; }
        public string RootCauseFilter { get; set; }
    }

    public class CorrectiveActionListModel
    {
        public int CorrectiveActionId { get; set; }
        public string Title { get; set; }
        public string AssignedBy { get; set; }
        public string CategoryName { get; set; }
        public string ActionSource { get; set; }
        public string ActionTaken { get; set; }
        public string DueDate { get; set; }
        public string RootCause { get; set; }
        public Nullable<long> RowID { get; set; }
    }

    public class CorrectiveActionList
    {
        public CorrectiveActionList()
        {
            this.CorrectiveActionListModel = new List<Models.CorrectiveActionListModel>();
        }
        public List<CorrectiveActionListModel> CorrectiveActionListModel { get; set; }
        public int TotalRecords { get; set; }
    }
}
